crates/bevy_scene/src/hook.rs

```diff
@@ -0,0 +1,174 @@
+//! Systems to insert components on loaded scenes.
+//!
+//! Please see the [`SceneHook`] documentation for detailed examples.
+
+use bevy_ecs::{
+    component::Component,
+    entity::Entity,
+    prelude::{Without, World},
+    system::{Commands, EntityCommands, Query, Res},
+    world::EntityRef,
+};
+
+use crate::{SceneInstance, SceneSpawner};
+
+/// Marker Component for scenes that were hooked.
+#[derive(Component)]
+#[non_exhaustive]
+pub struct SceneHooked;
```

Comment:
Note that this component is automatically loaded into all scenes added with `SceneBundle`, since `SceneHook` is a component of the `SceneBundle`.